images/virtualization-artifact/pkg/controller/vm/internal/sync_kvvm.go

```diff
@@ -97,13 +97,29 @@ func (h *SyncKvvmHandler) Handle(ctx context.Context, s state.VirtualMachineStat
 			Message(service.CapitalizeFirstLetter(err.Error()) + ".")
 		return reconcile.Result{}, err
 	}
+	class, err := s.Class(ctx)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
 	// 1. Set RestartAwaitingChanges.
-	var lastAppliedSpec *virtv2.VirtualMachineSpec
-	var changes vmchange.SpecChanges
+	var (
+		lastAppliedSpec *virtv2.VirtualMachineSpec
+		changes         vmchange.SpecChanges
+		allChanges      vmchange.SpecChanges
+		classChanged    bool
+	)
 	if kvvm != nil {
 		lastAppliedSpec = h.loadLastAppliedSpec(current, kvvm)
+		lastClassAppliedSpec := h.loadClassLastAppliedSpec(class, kvvm)
 		changes = h.detectSpecChanges(ctx, kvvm, &current.Spec, lastAppliedSpec)
+		if !changes.IsEmpty() {
+			allChanges.Add(changes.GetAll()...)
+		}
```

Comment:
I don't get it, why this copy "allChanges" is needed?

Reply:
lost `allChanges.Add(classChanges.GetAll()...)`
allChanges contains vmChanges + classChanges.
This is necessary to preserve the old logic for applying the VM configuration

Reply:
Oh, it make sense now.